### PR TITLE
Fix/support envvars for lookup aws ssm

### LINKED ISSUE #519: aws_ssm look does not support endpoint_url
### Summary

**what was tried**

using the [aws_ssm lookup](https://github.com/ansible-collections/amazon.aws/blob/main/plugins/lookup/aws_ssm.py) against a mock ssm parameter in localstack currently will not work.

```
my_cool_param: "{{ lookup('aws_ssm', '/namespace/my_cool_param' ) }}"
```

**proposed fix**

Add a configurable endpoint_url parameter into the [boto3 Session client](https://boto3.amazonaws.com/v1/documentation/api/latest/reference/core/session.html).

**ideal outcome**

1. an option that can be passed to the `aws_ssm` ansible lookup: 

```
my_cool_param: "{{ lookup('aws_ssm', '/namespace/my_cool_param', aws_endpoint_url=http://localhost:4566 ) }}"
```

2. An environment variable it also accept can accept

```
export AWS_ENDPOINT_URL=http://localhost:4566
```

### Issue Type

Feature Idea

### Component Name

aws_ssm lookup

### Additional Information

the new feature would allow ansible users to use a custom endpoint url rather than the default aws endpoints. 
localstack may be used then for example

```yaml (paste below)
my_cool_param: "{{ lookup('aws_ssm', '/namespace/my_cool_param', aws_endpoint_url=http://localhost:4566 ) }}"
```

### Code of Conduct

- [X] I agree to follow the Ansible Code of Conduct

### PR DESCRIPTION
**Pull request by Cicatrice**
Depends-On: https://github.com/ansible/ansible-zuul-jobs/pull/1424
##### SUMMARY

Working on support in `lookup` plugin `amazon.aws.aws_ssm` for `endpoint` parameters, as well of tradional environment variable for client configuration (`AWS_PROFILE`, `AWS_ACCESS_KEY_ID`, ...).

This should fixes #519 

Depends-On: https://github.com/ansible-collections/amazon.aws/pull/728

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
`aws_ssm`

##### ADDITIONAL INFORMATION

It seems `LookupModule` cannot benefit from `AnsibleAWSModule` (it expect several methods/params, I am not familiar enough with Python and Ansible to sort it out) ; so this PR uses `boto3_conn` and `get_aws_connection_info` defined in `utils.ec2`.

This is a simple snippet how it can be used:
```yaml
- name: Load env
  collections:
  - amazon.aws
  gather_facts: no
  hosts: localhost
  connection: local
  vars:
    my_env: "{{ lookup('amazon.aws.aws_ssm', 'status', endpoint='http://localhost:4566') }}"
  tasks:
    - name: show the env
      debug:
        msg: "{{ my_env }}"

```

Command line test:
```bash
ANSIBLE_COLLECTIONS_PATHS=${ANSIBLE_HOME}/collections ansible-playbook test.yml
```

Output:
```
PLAY [Load env] *************************************************************************************************************************************************************************************************************************************************************

TASK [show the env] *********************************************************************************************************************************************************************************************************************************************************
ok: [localhost] => {
    "msg": "INIT"
}

PLAY RECAP ******************************************************************************************************************************************************************************************************************************************************************
localhost                  : ok=1    changed=0    unreachable=0    failed=0    skipped=0    rescued=0    ignored=0   
```

Another test with envars:
```bash
AWS_ACCESS_KEY_ID='foo' AWS_ACCESS_SECRET_KEY='bar' AWS_URL='http://localhost:4566' ANSIBLE_COLLECTIONS_PATHS=${ANSIBLE_HOME}/collections ansible-playbook test.yml
```
where we remove the `endpoint` from `lookup`:
```yaml
- name: Load env
  collections:
  - amazon.aws
  gather_facts: no
  hosts: localhost
  connection: local
  vars:
    my_env: "{{ lookup('amazon.aws.aws_ssm', 'status') }}"
  tasks:
    - name: show the env
      debug:
        msg: "{{ my_env }}"
```

(provides same output)

### CODE REVIEW

**Comment by Cicatrice** [comment]
recheck

**Comment by Cicatrice** [comment]
recheck

**Comment by Cicatrice** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

File: plugins/lookup/aws_ssm.py [hunk 1]
```diff
@@ -74,6 +74,9 @@
     type: string
     choices: ['error', 'skip', 'warn']
     version_added: 2.0.0
+  endpoint:
+    description: Use a custom endpoint when connecting to SSM service
+    type: string
```
Comment (markuman) [inline plugins/lookup/aws_ssm.py:79]:
```suggestion
    type: string
    version_added: 3.2.0
```

**Comment by markuman** [comment]
recheck

**Comment by markuman** [comment]
@Cicatrice can you also add a channgelog fragment please?

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by alinabuzachis** [comment]
recheck

**Comment by abikouo** [comment]
recheck

**Comment by alinabuzachis** [comment]
@Cicatrice Could you please rebase your branch?

**Comment by alinabuzachis** [comment]
@Cicatrice Could you please rebase your branch?

File: plugins/lookup/aws_ssm.py (hunk 1 above)
Comment (alinabuzachis) [reply plugins/lookup/aws_ssm.py:79]: `version_added: 3.4.0`

**Comment by alinabuzachis** [comment]
@Cicatrice could you also please add a changelog fragment for this fix https://cn-ansibledoc.readthedocs.io/zh_CN/latest/community/development_process.html#changelogs? Thank you.

**Comment by alinabuzachis** [comment]
@Cicatrice could you also please add a changelog fragment for this fix https://cn-ansibledoc.readthedocs.io/zh_CN/latest/community/development_process.html#changelogs? Thank you.

**Review by tremble — CHANGES_REQUESTED** [review]
Thanks for taking the time to submit this PR.
2 minor things:

1) Please add a changelog fragment https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to
2) boto3_conn already catches the various botocore exceptions, there's no need to re-catch them.

**Pull request closed without merging**